In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load 
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()




FileNotFoundError: [Errno 2] No such file or directory: 'Resources/purchase_data.csv'

In [47]:
# Calculate total number of players
total_players = len(purchase_data['SN'].unique())
total_players_df = pd.DataFrame({'Total Players': [total_players]})
total_players_df


Total Players
0            576

In [48]:
Items = purchase_data["Item Name"].nunique()
average_price = round(purchase_data["Price"].mean(), 2)
number_of_purchases = purchase_data["Item Name"].nunique()
total_revenue = purchase_data["Price"].sum()

purchase_summary_table = pd.DataFrame({"Number Unique Items": [Items],
                                      "Average Price": [average_price],
                                      "Number Of Purchases": [number_of_purchases],
                                      "Total Revenue": [total_revenue]
                                      })

purchase_summary_table['Average Price'] = purchase_summary_table['Average Price'].map('${:.2f}'.format)
purchase_summary_table['Total Revenue'] = purchase_summary_table['Total Revenue'].map('${:,.2f}'.format)

purchase_summary_table

Number Unique Items Average Price  Number Of Purchases Total Revenue
0                  179         $3.05                  179     $2,379.77

In [49]:
gender_group = purchase_data.groupby('Gender')
gender_s = gender_group['SN'].nunique().sort_values(ascending=False)
gender_df = pd.DataFrame(gender_s)


gender_df = gender_df.rename(columns={'SN': 'Total Count'})
gender_df['Percentage of Players'] = ((gender_df['Total Count'] / total_players) * 100).map('{:.2f}%'.format)
gender_df.index.name=None
gender_df


Total Count Percentage of Players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [57]:
purchase_sum = purchase_data.groupby(["Gender"]).sum()["Price"]
purchase_avg = purchase_data.groupby(["Gender"]).mean()["Price"]
purchase_counts = purchase_data.groupby(["Gender"]).count()["Price"]


purchase_avg = purchase_sum / purchase_data.groupby(["Gender"]).nunique()["SN"]

gender_data = pd.DataFrame({"Purchase Count": purchase_counts, 
                            "Average Purchase Price": purchase_avg.map("${:.2f}".format),
                            "Total Purchase Value": purchase_sum.map("${:.2f}".format),
                            "Avg Total Purchase per Person": Avg_p.map("${:.2f}".format)})
gender_df.index.name=None
gender_data

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $4.47   
Male                              652                  $4.07   
Other / Non-Disclosed              15                  $4.56   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [51]:
age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]


purchase_data["Age Ranges"] = pd.cut(purchase_data["Age"], age_bins, labels=group_names)

age_demographics_totals = purchase_data.groupby(["Age Ranges"]).nunique()["SN"]
age_demographics_percents = age_demographics_totals / purchase_data['SN'].nunique() * 100

age_demographics = pd.DataFrame({"Total Count": age_demographics_totals, "Percent of Players": age_demographics_percents})

age_demographics

Total Count  Percent of Players
Age Ranges                                 
<10                  17            2.951389
10-14                22            3.819444
15-19               107           18.576389
20-24               258           44.791667
25-29                77           13.368056
30-34                52            9.027778
35-39                31            5.381944
40+                  12            2.083333

In [52]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_df = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total.map("${:.2f}".format)})

item_df.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [53]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_data = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total.map("${:.2f}".format)})

item_data.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [54]:
item_count = purchase_data.groupby(["Item ID","Item Name"]).count()["Price"]
item_total = purchase_data.groupby(["Item ID","Item Name"]).sum()["Price"]
item_average = purchase_data.groupby(["Item ID","Item Name"]).mean()["Price"]

item_data = pd.DataFrame({"Purchase Count": item_count,
                          "Item Price": item_average.map("${:.2f}".format),
                          "Total Purchase Value": item_total.map("${:.2f}".format)})

item_data.sort_values("Purchase Count", ascending=False).head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [55]:
 item_data.sort_values("Total Purchase Value", ascending=False ).head()

Purchase Count Item Price  \
Item ID Item Name                                                      
63      Stormfury Mace                                  2      $4.99   
29      Chaos, Ender of the End                         5      $1.98   
173     Stormfury Longsword                             2      $4.93   
38      The Void, Vengeance of Dark Magic               4      $2.37   
143     Frenzied Scimitar                               6      $1.56   

                                          Total Purchase Value  
Item ID Item Name                                               
63      Stormfury Mace                                   $9.98  
29      Chaos, Ender of the End                          $9.90  
173     Stormfury Longsword                              $9.86  
38      The Void, Vengeance of Dark Magic                $9.48  
143     Frenzied Scimitar                                $9.36

Observable trends:

• Gender purchase analysis: 84% of players are males (484) and only 14% of players are females (81). The average total purchase per person is almost the same ( 4.47𝑓𝑜𝑟𝑚𝑎𝑙𝑒𝑎𝑛𝑑 4.07 for female)

• Age demographics purchase analysis: 44.79% of players are between 20 and 24 years old, followed by 18.57% between the age of 15 and 19 years old. The players between the age of 25 and 29 represent 13.36 of the players.

• Most popular items and most profitable items: The most popular item is Final Critic with 13 purchases followed by Oathbreaker, Last Hope of the Breaking Storm with 12 purchases. These 2 items are the most profitable too followed in the third place by Nirvana.

